mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -1297,28 +1331,44 @@ void DeleteMember (string reason, string output, XmlNode member, MyXmlNodeList t
 				member.OwnerDocument.DocumentElement.GetAttribute ("FullName"),
 				member.Attributes ["MemberName"].Value, 
 				member.SelectSingleNode ("MemberSignature[@Language='C#']/@Value").Value);
-			if (!delete && MemberDocsHaveUserContent (member)) {
-				Warning ("Member deletions must be enabled with the --delete option.");
-			} else if (HasDroppedNamespace (type)) {
-				// if we're dropping the namespace, add the "classic style"
-				var existingAttribute = member.Attributes ["apistyle"];
-				if (existingAttribute != null) {
-					existingAttribute.Value = "classic";
-				} else {
-					// add the attribute and do not remove
-					XmlAttribute apistyleAttr = member.OwnerDocument.CreateAttribute ("apistyle");
 
-					apistyleAttr.Value = "classic";
+		// Identify all of the different states that could affect our decision to delete the member
+		bool shouldPreserve = !string.IsNullOrWhiteSpace (PreserveTag);
+		bool hasContent = MemberDocsHaveUserContent (member);
+		bool shouldDelete = !shouldPreserve && ((delete && !hasContent) || (!delete && !hasContent));
 
-					member.Attributes.Append (apistyleAttr);
-				}
-			} else if (!HasDroppedNamespace (type) && member.Attributes ["apistyle"] != null && member.Attributes ["apistyle"].Value == "unified") {
-				// do nothing if there's an apistyle=new attribute and we haven't dropped the namespace
-			} else if (!string.IsNullOrWhiteSpace (PreserveTag)) {
-				// do nothing
-			} else {
+		bool unifiedRun = HasDroppedNamespace (type);
+
+		var classicAssemblyInfo = member.SelectSingleNode ("AssemblyInfo[@apistyle='classic']");
+		bool nodeIsClassic = classicAssemblyInfo != null;
+
+		Action actuallyDelete = () => {
 			todelete.Add (member);
 			deletions++;
+		};
+
+		if (!shouldDelete) {
+			// explicitly not deleting
+			string message = shouldPreserve ? 
+				"Not deleting due to --preserve." :
+				"Member deletions must be enabled with the --delete option";
+			Warning (message);
+		} else if (unifiedRun && nodeIsClassic) {
+			// this is a unified run, and the member doesn't exist, but is marked as being in the classic assembly.
+			Warning ("Not removing this node since it's still in the classic assembly.");
+		} else if (unifiedRun && !nodeIsClassic) {
+			// unified run, and the node is not classic, which means it doesn't exist anywhere.
+			actuallyDelete ();
+		} else {
+			if (!nodeIsClassic) {
```

Comment:
This could be rephrased as:

    } else if (unifiedRun && !nodeIsClassic) {
        actuallyDelete();
    } if (!nodeIsClassic) {
        actuallyDelete();
    } else {
        Warning(...);
    }

Right?

Which in turn can be simplified to:

    } else if (!nodeIsClassic) {
        actuallyDelete ();
    }

because we don't appear to actually care if it's unified or not, just that it's not classic, no?